File: go/dsa/algorithms/arrays/two-sum-i-i.go
func twoSum(numbers []int, target int) []int {
    // 🟠 INITIALIZE: Start pointers at opposite ends
    left, right := 0, len(numbers) - 1

    // 🟢 LOOP & PROCESS: Converge until match or pointers meet
    for left < right {
        // 1️⃣ Process: Evaluate the current state
        currentSum := numbers[left] + numbers[right]
        
        if currentSum == target {
            return []int{left + 1, right + 1}
        }

        // 2️⃣ Decide and Act: Shrink search space
        if currentSum < target {
            left++ // Sum too small, move left inward
        } else {
            right-- // Sum too large, move right inward
        }
    }
    return []int{}
}

File: cpp/dsa/algorithms/arrays/two-sum-i-i.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // 🟠 INITIALIZE: Start pointers at opposite ends
        int left = 0, right = numbers.size() - 1;

        // 🟢 LOOP & PROCESS: Converge until match or pointers meet
        while (left < right) {
            // 1️⃣ Process: Evaluate the current state
            int currentSum = numbers[left] + numbers[right];
            
            if (currentSum == target) {
                return {left + 1, right + 1};
            }

            // 2️⃣ Decide and Act: Shrink search space
            if (currentSum < target) {
                left++; // Sum too small, move left inward
            } else {
                right--; // Sum too large, move right inward
            }
        }
        return {};
    }
};